src/main/java/org/opensearch/ad/ml/ThresholdingResult.java

```diff
@@ -59,20 +99,97 @@ public double getRcfScore() {
+        return totalUpdates;
+    }
+
+    public boolean isStartOfAnomaly() {
+        return startOfAnomaly;
+    }
+
+    public boolean isInHighScoreRegion() {
+        return inHighScoreRegion;
+    }
+
+    public int getRelativeIndex() {
+        return relativeIndex;
+    }
+
+    public double[] getCurrentTimeAttribution() {
+        return currentTimeAttribution;
+    }
+
+    public double[] getOldValues() {
+        return oldValues;
+    }
+
+    public double[][] getExpectedValuesList() {
+        return expectedValuesList;
+    }
+
+    public double getThreshold() {
+        return threshold;
+    }
+
+    public int getForestSize() {
+        return forestSize;
+    }
+
     @Override
     public boolean equals(Object o) {
         if (this == o)
             return true;
         if (o == null || getClass() != o.getClass())
             return false;
         ThresholdingResult that = (ThresholdingResult) o;
-        return Objects.equals(this.grade, that.grade)
-            && Objects.equals(this.confidence, that.confidence)
-            && Objects.equals(this.rcfScore, that.rcfScore);
+        return this.grade == that.grade
+            && this.confidence == that.confidence
+            && this.rcfScore == that.rcfScore
+            && this.totalUpdates == that.totalUpdates
+            && this.startOfAnomaly == that.startOfAnomaly
+            && this.inHighScoreRegion == that.inHighScoreRegion
+            && this.relativeIndex == that.relativeIndex
+            && Arrays.equals(currentTimeAttribution, currentTimeAttribution)
+            && Arrays.equals(oldValues, oldValues)
+            && Arrays.deepEquals(expectedValuesList, expectedValuesList)
+            && threshold == that.threshold
+            && forestSize == that.forestSize;
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(grade, confidence, rcfScore);
+        return Objects
+            .hash(
+                grade,
+                confidence,
+                rcfScore,
+                totalUpdates,
+                startOfAnomaly,
+                inHighScoreRegion,
+                relativeIndex,
+                Arrays.hashCode(currentTimeAttribution),
+                Arrays.hashCode(oldValues),
+                Arrays.deepHashCode(expectedValuesList),
+                threshold,
+                forestSize
+            );
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+            .append("grade", grade)
+            .append("confidence", confidence)
+            .append("rcfScore", rcfScore)
+            .append("totalUpdates", totalUpdates)
+            .append("startOfAnomaly", startOfAnomaly)
+            .append("inHighScoreRegion", inHighScoreRegion)
+            .append("relativeIndex", relativeIndex)
+            .append("currentTimeAttribution", Arrays.toString(currentTimeAttribution))
+            .append("oldValues", Arrays.toString(oldValues))
```

Comment:
This is changing right?

Reply:
I saw the old -> past -> relevant discussion ..

Reply:
yes, changed to pastValues

Reply:
> I saw the old -> past -> relevant discussion ..

Yes, not an easy choice.